<!DOCTYPE html>
<html lang='en'>
  <head>
    <meta charset='utf-8'>
    <meta content='width=device-width, initial-scale=1.0' name='viewport'>
    <meta content='ie=edge' http-equiv='x-ua-compatible'>
    <title>Bioinformatics</title>
    <meta content='Bioinformatics' name='description'>
    <meta content='panayiotis' name='author'>
    <link href='/bioinformatics/assets/icon.png' rel='icon'>
    <link href='/bioinformatics/assets/reveal.css' rel='stylesheet'>
    <link href='/bioinformatics/assets/black.css' rel='stylesheet'>
    <script src='/bioinformatics/assets/reveal.js' type='text/javascript'></script>
  </head>
  <body>
    <style>
      img { border: none !important; background: none !important;box-shadow: none !important;}
    </style>
    <div class='reveal'>
      <div class='slides'>
        <section>
          <h2>Genomic Processing Pipeline</h2>
          
          <h2>for</h2>
          
          <h2>population stratification analysis</h2>
        </section>
        <section>
          <span style='font-size: 4rem;'>
            <h3>Lets talk about...</h3>
            
            <ol>
            <li>Data</li>
            <li>Algorithm</li>
            <li>Application Stack</li>
            </ol>
          </span>
        </section>
        <section>
          <section>
            <h1>Data</h1>
          </section>
          <section>
            <h1>1 genome</h1>
            
            <p>3 billion char sequence, ~700 megabytes</p>
            
            <p>right off the genome sequencer, ~250 gigabytes</p>
            
            <p>1 per 1000 base is variable in human population</p>
            
            <p>all humans are 99.5% similar to any other humans</p>
          </section>
          <section>
            <h1>Variant calling</h1>
            
            <p>99.5% of genome data is shared across the genomes</p>
            
            <p>By using the variant call format only the variations
            need to be stored along with a reference genome.</p>
            
            <p>In VCF format a genome takes up ~125 megabytes</p>
          </section>
          <section>
            <h1>1000genomes.org</h1>
            
            <p>launched in January 2008,
            was an international research effort to establish by far
            the most detailed catalogue of human genetic variations
            including SNPs, structural variants, and their haplotype context.
            The
            <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/">final data set</a>
            contains data for 2,504 individuals from 26 populations.</p>
          </section>
          <section>
            <h2>Map</h2>
            <img src='/bioinformatics/assets/1000_genomes_map.png'>
          </section>
          <section>
            <h3>The 100,000 Genomes Project</h3>
            <img src='/bioinformatics/assets/genomics_england.png'>
            <p>The project will sequence 100,000 genomes from around 70,000 people.
            Participants are patients with a rare disease,
            plus their families, and patients with cancer.</p>
          </section>
        </section>
        <section>
          <section>
            <h1>Algorithm</h1>
          </section>
          <section>
            <h2>population stratification analysis</h2>
            
            <p>We are given a genome and we want
            to find matching subpopulations</p>
          </section>
          <section>
            <h2>Machine Learning</h2>
            
            <p>run a machine learning algorithm on a massive number of genomes with known 
            geographic origin.</p>
            
            <ul>
            <li>Find clusters</li>
            <li>Train Neural networks</li>
            <li>Build classifiers</li>
            </ul>
          </section>
          <section>
            <h2>Evaluation</h2>
            
            <ul>
            <li>split the data into a training and a test set</li>
            <li>run the algorithm on the training set</li>
            <li>evaluate the algorithm with the test set</li>
            </ul>
          </section>
        </section>
        <section>
          <section>
            <h1>Application Stack</h1>
          </section>
          <section>
            <img src='/bioinformatics/assets/spark.png'>
            <p>a fast and general engine for large-scale data processing.</p>
            
            <ul>
            <li>MapReduce cluster computing paradigm</li>
            <li>functional model of programming</li>
            </ul>
          </section>
          <section>
            <h1>ADAM</h1>
            
            <p>Current genomics data formats and processing
            pipelines are not designed to scale well to large
            datasets.</p>
            
            <ul>
            <li>read write genomic data operations</li>
            <li>data compression
            
            <ul>
            <li>.vcf.gz 210MB</li>
            <li>.vcf 11GB</li>
            <li>.adam 280MB</li>
            </ul></li>
            </ul>
          </section>
          <section>
            <h1>IAAS</h1>
            
            <ul>
            <li>elasticity</li>
            <li>scalability</li>
            </ul>
          </section>
        </section>
      </div>
    </div>
    <script>
      (function() {
        Reveal.initialize({
          history: true
        });
      
      }).call(this);
    </script>
  </body>
</html>
